src/Database/LSMTree/Internal.hs

```diff
@@ -1394,15 +1394,13 @@ listSnapshots sesh = do
       snaps <- mapM (checkSnapshot hfs root) $ Set.toList contents
       pure $ catMaybes snaps
   where
-    checkSnapshot hfs root s =
-      case Paths.mkSnapshotName s of
-        Nothing   -> pure Nothing
-        Just snap -> do
-          -- check that it is a directory
-          b <- FS.doesDirectoryExist hfs
-                (Paths.getNamedSnapshotDir $ Paths.namedSnapshotDir root snap)
-          if b then pure $ Just snap
-               else pure $ Nothing
+    checkSnapshot hfs root s = do
+      let snap = Paths.toSnapshotName s
+      -- check that it is a directory
+      b <- FS.doesDirectoryExist hfs
+            (Paths.getNamedSnapshotDir $ Paths.namedSnapshotDir root snap)
+      if b then pure $ Just snap
+            else pure $ Nothing
```

Comment:
This is not doing the same thing as before, is that okay? Should `listSnapshots` throw an error if there is a file/directory that violates `isValidSnapshotname`, or should it ignore the file/directory?